src/pkg/cluster/tunnel.go

```diff
@@ -267,42 +267,42 @@ func (c *Cluster) findPodContainerPort(ctx context.Context, svc corev1.Service)
 }
 
 // TODO: Refactor to use netip.AddrPort instead of a string for nodePortURL.
-func serviceInfoFromNodePortURL(services []corev1.Service, nodePortURL string) (string, string, int, error) {
+func serviceInfoFromNodePortURL(services []corev1.Service, nodePortURL string) (corev1.Service, int, error) {
```

Comment:
What is the purpose for changing the function signature?

Reply:
I need more information from the service, rather than returning yet another string it felt easier to just return a service